components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/matchers/token/TokenParser.java

```diff
@@ -84,7 +85,16 @@ public TokenParser(final String templatePattern) {
 
     Map<String, String> values = new HashMap<>();
     for (int i = 0; i < matcher.groupCount(); i++) {
-      values.put(variables.get(i).getName(), matcher.group(i + 1));
+      if (values.containsKey(variables.get(i).getName())) {
+        if (!Objects.equals(values.get(variables.get(i).getName()), matcher.group(i + 1))) {
+          log.trace("Variable '{}' values mismatch: '{}' vs '{}'", variables.get(i).getName(),
+              values.get(variables.get(i).getName()), matcher.group(i + 1));
```

Comment:
This change is a good idea, but now I regret not refactoring out a couple of local variables in my original code. I need more coffee to parse this. :)  What about:
    final String varName = variables.get(i).getName();
    final String value = matcher.group(i + 1);